modules/mobile/app/models/mobile/v0/adapters/immunizations.rb

```diff
@@ -36,6 +36,11 @@ def sort_by_date_and_group_name(vaccine_map)
 
             [[date_sort_key1, date_sort_key2], immunization[:group_name]]
           end
+        rescue ArgumentError
```

Comment:
This is what the ticket requests you do, but now it will result in another downstream error because this will probably just return nil and then we'll have an "operating on nil" error. Might be better to explicitly raise here. 

But I also believe the underlying issue here is that there's an immunization record without a group name. I would consider writing tests for "no group name" and determining what the behavior should be. See screenshot for why I think that's the issue. 

![image](https://user-images.githubusercontent.com/28661253/192357205-47666ee3-a022-454e-843b-2e991d7b098e.png)

Reply:
To be clear, I'm not saying we shouldn't log the error. I'm just saying we might be better off fixing the obvious case of "when there is no group name". We can still log and then maybe raise an explicit 422. 

Also, FWIW, I don't think we need to worry about personal information in the logging. I believe it will only be available via datadog, which even we are struggling to get access to.

Reply:
I initially thought the same but I couldn't get `comparison of Array with Array` to be raised. In `Mobile::V0::Adapters::Immunizations`, group name gets set by this:
```
        def group_name(vaccine_code)
          group_name = vaccine_code.dig(:coding, 1, :display)
          group_name&.slice!('VACCINE GROUP: ')
          group_name.presence
        end
```
Which will leave the immunization instance with `group_name: nil`. Unlike your screenshot, the key `:group_name` will still exist, so the comparison `[[date_sort_key1, date_sort_key2], immunization[:group_name]]` will actually still be a valid comparison.

I added the specs as you suggested with missing group names. I didn't write more in depth specs until the error is found and we can then decide how the array should be ordered after the failure situation.

Reply:
I see what you mean. My console test didn't accurately capture the issue. But I tried again with group_name: nil and got the same result. This might be easier to test with unit tests instead of request specs. If you're up for it, HMU when you're online and we can pair on this for a few minutes today.

Reply:
Ok figured out the issue.
```
[[date_sort_key1, date_sort_key2], immunization[:group_name]]
```

`immunization[:group_name]` will only be compared if the first arrays are equal. The cassette needed to be changed so that two immunizations have the same date AND one of those two has a `nil` group_name. After doing so, the spec was actually causing the necessary issue. After that, I corrected the issue in `sort_by_date_and_group_name` and removed logging since there should not be a failure state any more